File: Python/easy/169. Majority Element.py
"""
Given an array of size n, find the majority element. The majority element is the element that appears more than ⌊ n/2 ⌋ times.
You may assume that the array is non-empty and the majority element always exist in the array.
Example 1:
Input: [3,2,3]
Output: 3
Example 2:
Input: [2,2,1,1,1,2,2]
Output: 2
"""
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        appearDict = {}
        for i in nums:
            if i not in appearDict:
                appearDict[i] = 1
            else:
                appearDict[i] += 1
        for i in appearDict:
            if appearDict[i] > len(nums) / 2:
                return i

File: Java/easy/169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int result = 0;
        int counter = 0;
        for (int i = 0; i < nums.length; i++) {
            if (counter == 0) {
                result = nums[i];
            }
            counter += result == nums[i] ? 1 : -1;
        }
        return result;
    }
}